test/widgets/compose_box_test.dart

```diff
@@ -155,13 +172,71 @@ void main() {
     }
 
     testWidgets('_StreamComposeBox', (tester) async {
-      await prepareComposeBox(tester, StreamNarrow(eg.stream().streamId));
-      checkComposeBoxTextFields(tester, expectTopicTextField: true);
+      final key = await prepareComposeBox(tester,
+        StreamNarrow(eg.stream().streamId));
+      checkComposeBoxTextFields(tester, controllerKey: key,
+        expectTopicTextField: true);
     });
 
     testWidgets('_FixedDestinationComposeBox', (tester) async {
-      await prepareComposeBox(tester, TopicNarrow.ofMessage(eg.streamMessage()));
-      checkComposeBoxTextFields(tester, expectTopicTextField: false);
+      final key = await prepareComposeBox(tester,
+        TopicNarrow.ofMessage(eg.streamMessage()));
+      checkComposeBoxTextFields(tester, controllerKey: key,
+        expectTopicTextField: false);
+    });
+  });
+
+  group('message-send request response', () {
+    Future<void> setupAndTapSend(WidgetTester tester, {
+      required void Function(int) prepareResponse,
```

Comment:
nit: can give a hint what the callback parameter means:
```suggestion
      required void Function(int messageId) prepareResponse,
```

(wasn't so necessary when the type was Message)